presto-main/src/main/java/com/facebook/presto/operator/TaskResultClient.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.operator.PageBufferClient.PagesResponse;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.DataSize;
+
+/**
+ * All methods in this class should be async
+ */
+public interface TaskResultClient
+{
+    ListenableFuture<PagesResponse> getResults(long token, DataSize maxResponseSize);
+
+    /**
+     * A fire and forget call to issue the ack to the buffer.
+     * No need to handle the response; it is ok for a server to miss the ack.
+     * {@param nextToken} N represents token N - 1 is to be acknowledged
+     */
+    void acknowledgeResults(long nextToken);
+
+    ListenableFuture<?> abortResults();
```

Comment:
Maybe explain why there can be case we no longer need result? (e.g. the later stage cancelled or finished with partial data, e.g. LIMIT? )

Reply:
Actually `abortResults` is only to close the remote buffer; while `deleteTask` (which is not covered by this PR) is to handle LIMIT cases.